pkg/apis/provisioning/v1alpha5/kubelet_configuration.go

```diff
@@ -22,4 +22,8 @@ type KubeletConfiguration struct {
 	// Note that not all providers may use all addresses.
 	//+optional
 	ClusterDNS []string `json:"clusterDNS,omitempty"`
+
+	// serviceClusterIPRange is a IP CIDR that will be used for assigning IP to services created on Kubernetes.
+	//+optional
+	ServiceClusterIPRange string `json:"serviceClusterIPRange,omitempty"`
```

Comment:
I add a Provisioner API's attribute for specifying API server's `--service-cluster-ip-range` parameter.
EKS optimized AMI uses the parameter to infer Cluster DNS Service IP.

Reply:
Why do we need the service-cluster-ip-range, if we can just specify the cluster-dns ip directly?

Reply:
We can specify only `service-cluster-ip-range` parameter without `cluster-dns`, however current implementation need `service-cluster-ip-range` even if `cluster-dns` is specified when using IPv6 cluster.
It's caused the EKS Optimized AMI's implementation in which `SERVICE_IPV6_CIDR` is mandatory:
https://github.com/awslabs/amazon-eks-ami/blob/v20220123/files/bootstrap.sh#L303

Reply:
If this is the case, we should fix the AMI, rather than polluting karpenter with additional complexity.

Reply:
Thank you for the feedbacks. I changed the PR to be able to use IPv6 provisioning node only when users specify the IPv6 DNS Cluster IP.
It may not so straightforward however we can use IPv6 with Karpenter without changing the Karpenter API.